lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,148 @@ def post(self, request, *args, **kwargs):
+        media: (dict) An object that contains named media items. Included here:
+            * course_image: An image to show for the course.  Represented
+              as an object with the following fields:
+                * uri: The location of the image
+        certificate: (dict) Information about the user's earned certificate in the course.
+            Included here:
+                * uri: The location of the user's certificate
+        is_self_paced: (bool) Indicates if the course is self paced
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent or the username was not provided
+        * 401 unauthorized, the provided access token has expired and is no longer valid
+          for an authenticated request.
+        * 403 if a user who does not have permission to masquerade as
+          another user specifies a username other than their own.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    def get_certificate(self, request, course_id):
+        """Returns the information about the user's certificate in the course."""
+        if request.user.is_authenticated:
+            certificate_info = certificate_downloadable_status(request.user, course_id)
+            if certificate_info['is_downloadable']:
+                return {
+                    'url': request.build_absolute_uri(
+                        certificate_info['download_url']
+                    ),
+                }
+        return {}
+
+    @staticmethod
+    def compose_course_info(course_overview):
+        """
+        Method for obtaining additional information about the course.
+
+        Arguments:
+            request - Django request object
+        """
+
+        course_data = {
+            # identifiers
+            'name': course_overview.display_name,
+            'number': course_overview.display_number_with_default,
+            'org': course_overview.display_org_with_default,
+
+            # dates
+            'start': course_overview.start,
+            'start_display': course_overview.start_display,
+            'start_type': course_overview.start_type,
+            'end': course_overview.end,
+
+            # various URLs
+            'media': {
+                'image': course_overview.image_urls,
+            },
+            'is_self_paced': course_overview.self_paced
+        }
+        return course_data
+
+    def list(self, request, **kwargs):  # pylint: disable=W0221
+        """
+        REST API endpoint for listing all the blocks information in the course and
+        information about the course while regarding user access and roles.
+
+        Arguments:
+            request - Django request object
+        """
+
+        response = super().list(request, kwargs)
+
+        if request.GET.get('return_type', 'dict') == 'dict':
+            course_id = request.query_params.get('course_id', None)
```

Comment:
What if course_id is None?
Please handle all these cases in tests.

Reply:
Hi Jawad!
The `course_id` is a required field and validated in the parent class - here https://github.com/raccoongang/edx-platform/blob/master/lms/djangoapps/course_api/blocks/views.py#L305


Furthermore, all the test cases are covered because TestBlocksInfoInCourseView class is extending [TestBlocksInCourseView](https://github.com/raccoongang/edx-platform/blob/79cc07a46e977028014c12685a83b7c63348c7b8/lms/djangoapps/course_api/blocks/tests/test_views.py#L20) which has those test cases covered.